src/sidebar/store/test/use-store-test.js

```diff
@@ -0,0 +1,108 @@
+'use strict';
+
+const { mount } = require('enzyme');
+const { createStore } = require('redux');
+const { createElement } = require('preact');
+const { act } = require('preact/test-utils');
+
+const useStore = require('../use-store');
+const { $imports } = useStore;
+
+const initialState = { value: 10, otherValue: 20 };
+const reducer = (state = initialState, action) => {
+  if (action.type === 'INCREMENT') {
+    return { ...state, value: state.value + 1 };
+  } else if (action.type === 'INCREMENT_OTHER') {
+    return { ...state, otherValue: state.otherValue + 1 };
+  } else {
+    return state;
+  }
+};
+
+describe('useStore', () => {
+  let renderCount;
+  let testStore;
+  let TestComponent;
+
+  beforeEach(() => {
+    renderCount = 0;
+
+    // eslint-disable-next-line react/display-name
+    TestComponent = () => {
+      renderCount += 1;
+      const aValue = useStore(store => store.getState().value);
+      return <div>{aValue}</div>;
+    };
+
+    testStore = createStore(reducer);
+    $imports.$mock({
+      '../util/service-context': {
+        useService: name => (name === 'store' ? testStore : null),
+      },
+    });
+  });
+
+  afterEach(() => {
+    $imports.$restore();
+  });
+
+  it('returns result of `callback(store)`', () => {
+    const wrapper = mount(<TestComponent />);
+    assert.equal(wrapper.text(), '10');
+  });
+
+  it('re-renders when the store changes and result of `callback(store)` also changes', () => {
```

Comment:
I'm unclear on how this tests re-rendering, as it uses `update` to, I think?, force-render?

Reply:
Hmm, I think [the documentation](https://airbnb.io/enzyme/docs/api/ReactWrapper/update.html) is misleading. `wrapper.update()` updates the wrapper (Enzyme's view of the output) to match what has currently been rendered. It doesn't actually re-render the component itself (`wrapper.setProps(...)` does).

There is a [note in the changelog](https://github.com/airbnb/enzyme/blob/master/docs/guides/migration-from-2-to-3.md#for-mount-updates-are-sometimes-required-when-they-werent-before) of the last Enzyme release which explains in more detail.